src/tests/testClpMap.cc

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "SquidConfig.h"
+#include "testClpMap.h"
+#include "unitTestMain.h"
+
+#include <ctime>
+
+CPPUNIT_TEST_SUITE_REGISTRATION( testClpMap );
+
+class SquidConfig Config;
+
+void
+testClpMap::addData(TestMap &m, int count, int startWith, TestMap::Ttl ttl)
+{
+    for (auto j = startWith; j < startWith + count; ++j)
```

Comment:
See my other comments about keeping things as simple as possible: if one is a size_t and the other is an int, to add them a cast will be needed. And then we might need to signal the compiler that no, we are sure, there will be no overflows. I had actually implemented this, which would turn a two-lines helper function into a five-lines helper function and actually decrease intent clarity.
Could this be made more theoretically correct? Absolutely. Efficient? Of course. Is any of that worth? I contend not. Also, done is better than perfect and we are wasting way too much time on this.
Let me clarify the full context of my objective here:
- I want to use ClpMap for ipcache and fqdncache, as discussed in a different PR
- to do that I need to have const iterators on ClpMap to display cache contents on cachemanager
- to develop the  iterators I want to have higher confidence in the correctness of what I implement
- to have more confidence I want to have unit tests so I can do test-driven development
- to do that I need unit tests, which are currently not in

We are at step zero of what is an actually useful result; it's been two weeks since I started and we are picking nits. At this pace, when will we get to any meaningful result?

Reply:
I did take your comments into account when adding my support for Amos' change request. My current theory is that it is possible to write this code using correct types without casts and or other serious complications (despite your claims to the contrary). Let me know when it is a good time for me to try to test that theory. My attempt will not cost you (or me) much and may improve this code, addressing two valid (albeit minor) change requests.

> we are picking nits

I disagree that this is what the focus of the current change requests is. The important change requests are not nitpicking IMO. The vast majority of other change requests are trivially addressable and not worth discussing. I am happy to address nitpicking (or all) change requests if that helps.

> to have more confidence I want to have unit tests so I can do test-driven development

If the current tests give you that confidence, use them! You do not need this PR to be officially merged to use these tests.

> At this pace, when will we get to any meaningful result?

AFAICT, we already have meaningful result for those who want to run these tests. IIRC, there were no complex change requests, and PR delays were primarily on your side -- _you_ determine the speed of progress in this case. That situation may change, of course, but that how it looks to me right now, and, frankly, I envy your position.

FWIW, I am happy to assist with speeding this useful PR up. Just let me know when it is a good time for me to do so. Again, I would have done that already, but I did not want to create conflicts with your active work.

Reply:
Done in cd911ae. FWIW, I did not run into any of the [predicted problems](https://github.com/squid-cache/squid/pull/1262/files#r1111673865).